aten/src/ATen/core/jit_type.h

```diff
@@ -1896,27 +1933,31 @@ struct CAFFE2_API ClassType : public NamedType {
     return n.qualifiedName();
   }
 
+  void provideNewAttributeKind(AttributeKind attributeKind);
```

Comment:
nit: I think this name doesn't make it immediately clear that this function is basically the intenral impl of `addAttribute`. When I read "provide" I'm not really sure how it relates to the other stuff in this class, if that makes sense.